src/algebra/big_operators/fin.lean

```diff
@@ -28,4 +29,48 @@ lemma prod_filter_succ_lt {M : Type*} [comm_monoid M] {n : ℕ} (j : fin n) (v :
     ∏ j in univ.filter (λ i, j < i), v j.succ :=
 by rw [univ_filter_succ_lt, finset.prod_map, rel_embedding.coe_fn_to_embedding, coe_succ_embedding]
 
+@[to_additive]
+lemma prod_congr {M : Type*} [comm_monoid M] {a b : ℕ} (f : fin b → M) (h : a = b) :
+  ∏ (i : fin a), f (fin.cast h i) = ∏ (i : fin b), f i :=
+by { subst h, congr, ext, congr, ext, rw fin.coe_cast, }
+
+@[to_additive]
+lemma prod_univ_add {M : Type*} [comm_monoid M] {a b : ℕ} (f : fin (a+b) → M) :
```

Comment:
Can you move this next to prod_univ_succ?

Reply:
Would not it be better if the `fin` specific results about products and sums in `data/fintype/card.lean` were moved here in `algebra/big_operators/fin.lean` instead?

Reply:
Sure, that sounds like a good idea; but I expect that will be more work. If you want to do that, I'd recommend pausing work on this PR and making a new PR that _only_ moves the existing lemmas between files, without adding any new ones.

Reply:
Yes, I shall try to do that within a few days.